src/engine/GameObjects/MapNode.cxx

```diff
@@ -287,6 +287,19 @@ const MapNodeData &MapNode::getActiveMapNodeData() const
   return m_mapNodeData[Layer::TERRAIN];
 }
 
+void MapNode::setMapNodeData(std::vector<MapNodeData> mapNodeData)
+{
+  if (mapNodeData.size() == m_mapNodeData.size())
+  {
+    m_mapNodeData.swap(mapNodeData);
+  }
+  // updates the pointers to the tiles, after loading tileIDs from json
+  for (auto &it : m_mapNodeData)
```

Comment:
We only need to update if we are actually swapping

Reply:
the size check was unnecessary and we can't compare the vector we get (cause it's a deserialization object) with the existing vector. At least not without introducing more overhead, which would make it even slower.

I removed the check (it wasn't doing anything) and left it as is.